pkg/util.go

```diff
@@ -129,3 +179,568 @@ func (opt *etcdOptions) setCredentials(sh Shell, appBinding *appcatalog.AppBindi
 	sh.SetEnv(EnvEtcdPassword, string(secret.Data[EtcdPassword]))
 	return nil
 }
+
+func clearDir(dir string) error {
+	if err := os.RemoveAll(dir); err != nil {
+		return fmt.Errorf("unable to clean datadir: %v. Reason: %v", dir, err)
+	}
+	return os.MkdirAll(dir, os.ModePerm)
+}
+
+func (opt *etcdOptions) createRestorePvcs(pvcList []corev1.PersistentVolumeClaim) ([]corev1.PersistentVolumeClaim, error) {
+
+	if len(pvcList) == 0 {
+		err := errors.New("No pvc found for etcd statefulset")
+		return nil, err
+	}
+
+	restorePvcList := make([]corev1.PersistentVolumeClaim, 0)
+	inv, err := invoker.ExtractRestoreInvokerInfo(opt.kubeClient, opt.stashClient, opt.InvokerKind, opt.InvokerName, opt.namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceRequirement := pvcList[0].Spec.Resources
+
+	if opt.memReq != "" || opt.memLimit != "" {
+		resourceRequirement = corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse(opt.cpuLimit),
+				corev1.ResourceMemory: resource.MustParse(opt.memLimit),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse(opt.cpuReq),
+				corev1.ResourceMemory: resource.MustParse(opt.memReq),
+			},
+		}
+	}
+	if opt.storageClassName == "" {
+		opt.storageClassName = *pvcList[0].Spec.StorageClassName
+	}
+
+	for _, pvcs := range pvcList {
+		pvc := &corev1.PersistentVolumeClaim{
+
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      meta.ValidNameWithPrefix("restore", pvcs.Name),
+				Namespace: opt.namespace,
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes:      pvcs.Spec.AccessModes,
+				Resources:        resourceRequirement,
+				StorageClassName: &opt.storageClassName,
+			},
+		}
+
+		createdPVC, _, err := core_util.CreateOrPatchPVC(context.TODO(), opt.kubeClient, pvc.ObjectMeta, func(in *corev1.PersistentVolumeClaim) *corev1.PersistentVolumeClaim {
+			// Set RestoreSession as owner of the PVC so that it get deleted when the respective owner is deleted.
+			core_util.EnsureOwnerReference(&in.ObjectMeta, inv.OwnerRef)
+			in.Spec = pvc.Spec
+			return in
+		}, metav1.PatchOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		restorePvcList = append(restorePvcList, *createdPVC)
+	}
+	return restorePvcList, nil
+}
+
+func (opt *etcdOptions) createRestorePods(pvcList, restorePvcList []corev1.PersistentVolumeClaim) ([]corev1.Pod, []string, error) {
```

Comment:
This will be called from `restoreMember()` func. So, only create the pod to restore that particular member.